### PR TITLE
Change the way Xml content is formatted. 

### LINKED ISSUE #501: ArgumentException: '.', hexadecimal value 0x00, is an invalid character.
```
Message: [ArgumentException: '.', hexadecimal value 0x00, is an invalid character.]
 System.Xml.XmlEncodedRawTextWriter.InvalidXmlChar(Int32 ch, Char* pDst, Boolean entitize):0
 System.Xml.XmlEncodedRawTextWriter.WriteElementTextBlock(Char* pSrc, Char* pSrcEnd):467
 System.Xml.XmlEncodedRawTextWriter.WriteString(String text):67
 System.Xml.XmlWellFormedWriter.WriteString(String text):58
 System.Xml.XmlElement.WriteElementTo(XmlWriter writer, XmlElement e):86
 System.Xml.XmlDocument.WriteContentTo(XmlWriter xw):28
 Particular.ServiceInsight.Desktop.ExtensionMethods.XmlDocumentExtensions.GetFormatted(XmlDocument document) in c:\BuildAgent\work\7540559e45164f93\src\ServiceInsight\ExtensionMethods\XmlDocumentExtensions.cs:18
 Particular.ServiceInsight.Desktop.MessageViewers.XmlViewer.XmlMessageView.Display(String message) in c:\BuildAgent\work\7540559e45164f93\src\ServiceInsight\MessageViewers\XmlViewer\XmlMessageView.xaml.cs:35
 Caliburn.Micro.ViewAware.Caliburn.Micro.IViewAware.AttachView(Object view, Object context) in c:\Users\Foo\Documents\CodePlex\caliburnmicro\src\Caliburn.Micro.Silverlight\ViewAware.cs:78
 Caliburn.Micro.ViewModelBinder.<.cctor>b__3(Object viewModel, DependencyObject view, Object context) in c:\Users\Foo\Documents\CodePlex\caliburnmicro\src\Caliburn.Micro.Silverlight\ViewModelBinder.cs:196
 Caliburn.Micro.View.OnModelChanged(DependencyObject targetLocation, DependencyPropertyChangedEventArgs args) in c:\Users\Foo\Documents\CodePlex\caliburnmicro\src\Caliburn.Micro.Silverlight\View.cs:299
 System.Windows.DependencyObject.OnPropertyChanged(DependencyPropertyChangedEventArgs e):82
 System.Windows.FrameworkElement.OnPropertyChanged(DependencyPropertyChangedEventArgs e):8
 System.Windows.DependencyObject.NotifyPropertyChange(DependencyPropertyChangedEventArgs args):0
 System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType):1016
 System.Windows.DependencyObject.InvalidateProperty(DependencyProperty dp, Boolean preserveCurrentValue):82
 System.Windows.Data.BindingExpressionBase.Invalidate(Boolean isASubPropertyChange):68
 System.Windows.Data.BindingExpression.TransferValue(Object newValue, Boolean isASubPropertyChange):956
 System.Windows.Data.BindingExpression.Activate(Object item):303
 System.Windows.Data.BindingExpression.AttachToContext(AttachAttempt attempt):993
 System.Windows.Data.BindingExpression.MS.Internal.Data.IDataBindEngineClient.AttachToContext(Boolean lastChance):0
 MS.Internal.Data.DataBindEngine+Task.Run(Boolean lastChance):96
 MS.Internal.Data.DataBindEngine.Run(Object arg):114
 MS.Internal.Data.DataBindEngine.OnLayoutUpdated(Object sender, EventArgs e):7
 System.Windows.ContextLayoutManager.fireLayoutUpdateEvent():88
 System.Windows.ContextLayoutManager.UpdateLayout():870
 System.Windows.Interop.HwndSource.SetLayoutSize():736
 System.Windows.Interop.HwndSource.set_RootVisualInternal(Visual value):258
 System.Windows.Window.SetRootVisualAndUpdateSTC():6
 System.Windows.Window.SetupInitialState(Double requestedTop, Double requestedLeft, Double requestedWidth, Double requestedHeight):806
 System.Windows.Window.CreateSourceWindow(Boolean duringShow):290
 System.Windows.Window.ShowHelper(Object booleanBox):59
 Particular.ServiceInsight.Desktop.App.OnStartup(StartupEventArgs e) in c:\BuildAgent\work\7540559e45164f93\src\ServiceInsight\App.xaml.cs:57
 System.Windows.Application.<.ctor>b__1(Object unused):15
 System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs):123
 MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler):17
```

Raygun report https://app.raygun.io/dashboard/7gulz9/errors/877619590 has more details about the error

### PR DESCRIPTION
**Pull request by HEskandari**
Resolves #501

@Particular/serviceinsight-maintainers Please review. This changes the way the xml string is formatted and with that addresses the exception in #501. I have moved the formatting method to the ViewModel where I believe it belongs and also added a test (it is not easy to add tests for stuff in the View).

### CODE REVIEW

File: src/ServiceInsight/ExtensionMethods/XmlDocumentExtensions.cs [hunk 1]
```diff
@@ -1,24 +1,19 @@
...
 {
     using System.IO;
-    using System.Text;
     using System.Xml;
+    using System.Xml.Linq;
 
     public static class XmlDocumentExtensions
     {
-        public static string GetFormatted(this XmlDocument document)
+        public static string FormattAsXml(this string xmlContent)
```
Comment (johnsimons) [inline src/ServiceInsight/ExtensionMethods/XmlDocumentExtensions.cs]: @HEskandari too many `t`s, Formatt should be Format ?

Comment (HEskandari) [inline src/ServiceInsight/ExtensionMethods/XmlDocumentExtensions.cs]: Thanks @johnsimons I'll update.

**Comment by HEskandari** [comment]
@Particular/serviceinsight-maintainers So I've moved the formatting method into the VM and removed the extension method as agreed. This should be good now.

File: src/ServiceInsight/MessageViewers/XmlViewer/XmlMessageViewModel.cs [hunk 2]
```diff
@@ -55,5 +60,24 @@ public void Clear()
...
+            try
+            {
+                using (var stringReader = new StringReader(message))
+                using (var xmlReader = new XmlTextReader(stringReader))
+                {
+                    var xml = XDocument.Load(xmlReader);
+                    return xml.ToString();
+                }
+            }
+            catch (Exception)
```
Comment (adamralph) [inline src/ServiceInsight/MessageViewers/XmlViewer/XmlMessageViewModel.cs]: Aha! This is where the problem is fixed. Previously we were catching `XmlException`.

**Pull request merged**